src/main/scala/com/amazon/deequ/constraints/Constraint.scala

```diff
@@ -49,8 +50,8 @@ class ConstraintDecorator(protected val _inner: Constraint) extends Constraint {
       analysisResults: Map[Analyzer[_, Metric[_]], Metric[_]])
     : ConstraintResult = {
 
-    val result = _inner.evaluate(analysisResults)
-    ConstraintResult(this, result.status, result.message)
+    // constraint = this to keep toString of NamedConstraint
```

Comment:
?

Reply:
Should have put a better description there. Most of the constraints are `NamedConstraint`s. `_inner` in this case is an `AnalysisBasedConstraint` which doesn't have a pretty-printable name. To keep the name and toString method, we keep the outer type.

Will update the comment.